src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Interop/msquic.cs

```diff
@@ -0,0 +1,191 @@
+#pragma warning disable IDE0073
+//
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+//
+#pragma warning restore IDE0073
+
+using System;
+using System.Diagnostics;
+using System.Net.Quic;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Quic
+{
+    internal unsafe partial struct QUIC_BUFFER
+    {
+        public Span<byte> Span => new(Buffer, (int)Length);
+    }
+
+    internal partial class MsQuic
+    {
+        public static unsafe QUIC_API_TABLE* Open()
+        {
+            QUIC_API_TABLE* ApiTable;
+            int Status = MsQuicOpenVersion(2, (void**)&ApiTable);
+            ThrowIfFailure(Status);
+            return ApiTable;
+        }
+
+        public static unsafe void Close(QUIC_API_TABLE* ApiTable)
+        {
+            MsQuicClose(ApiTable);
+        }
+
+        public static void ThrowIfFailure(int status, string? message = null)
+        {
+            if (StatusFailed(status))
+            {
+                // TODO make custom exception, and maybe throw helpers
+                throw new MsQuicException(status, message);
+            }
+        }
+
+        public static bool StatusSucceeded(int status)
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
```

Comment:
This is old inefficient way to dynamically check for Windows. The proper way to check for Windows is using `OperatingSystem.IsWindows()`